etc/travis-scripts

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-8 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+set -eu
+set -o xtrace  # for ease of debugging travis builds
+shopt -s extglob
+
+
+export PATH="${PWD}/bin:${HOME}/fcm-master/bin:${HOME}/cylc-master/bin:$PATH"
+
+TMP_DOCS_DIR="${TMPDIR:-/var/tmp}/temp-docs-dir"
+DOCUMENTATION_FILES=!(doc|404*|CHANGES.md|_config.yml)
```

Comment:
Since (as briefly discussed in person) a ``bash -n travis-scripts`` vaildation is confused by this extended globbing line, consider appending a comment here identifying this as such (if I may be pedantic).

Reply:
No idea why `bash -n` doesn't like the extglobs, perhaps it doesn't support them. For proper linting of shell script use something like shellcheck.